### DISCUSSION TITLE
v94 patches

### DISCUSSION THREAD

**Discussion by uazo** [bromite/bromite#1415] [General]
@csagan5 my patchs for [v94](https://github.com/uazo/bromite/commits/v94) 
[building tests](https://github.com/uazo/bromite-buildtools/actions/runs/1269414118)

**Comment by csagan5** [comment]
I am still working on v93; for example there is a `java.lang.ClassNotFoundException: com.google.android.material.theme.MaterialComponentsViewInflater` which I would like to fix before release.

Edit: it happens also on Chromium, so I will probably let it be.

**Reply by csagan5** [reply]
Another issue I noticed: tab groups are now disabled by default, but it becomes effective only after a restart.

**Reply by csagan5** [reply]
And what about the size? v93 has now +20MB. Any idea what is causing that increase?

**Reply by uazo** [reply]
the chromium repo has a tool to analyze apk space usage, i will try to see the results

**Reply by csagan5** [reply]
If there is something new that we do not use, we should remove it.

**Comment by uazo** [comment]
do you plan to bring bromite to v94 in the near future? i have a patch (for v94) that closes some issue and i don't know whether to backport to v93 or wait

**Reply by csagan5** [reply]
Not immediately; up to you. The differences should not be big?

**Reply by csagan5** [reply]
This is a list of issues assigned to you, in case you are looking for features to pick up: #728 #1062 #1182 #1331

**Reply by uazo** [reply]
sorry, little free time ...

I am currently working on #1182, and in the meantime also on #1362 
since the patch is always the same, I also added support to tab recents in always incognito mode (with enable history on).
[here](https://github.com/uazo/bromite/commit/0b2375f032e1d7ebc9052a74d9466db4f320ea4c) version for v94.
that patch also has tests, as I'm learning to develop them too, harder than I thought.

I am currently compiling, then I run the tests and see if it is easy to bring the patch to v93.

**Reply by snrkl** [reply]
Does the discovered 0-day security flaws actively being exploited in the wild change the timelines for bringing v94 out?  Chromium 94 was pushed urgently on all the desktop platforms to address. I'm wondering if these also affected mobile?

Details are Here: https://chromereleases.googleblog.com/2021/09/stable-channel-update-for-desktop_30.html?m=1

**Reply by csagan5** [reply]
> sorry, little free time ...

As I mentioned before, no need to be sorry.

> I'm wondering if these also affected mobile?

If it were there would be a post: https://chromereleases.googleblog.com/search/label/Chrome%20for%20Android

**Reply by Zoraver** [reply]
>If it were there would be a post: https://chromereleases.googleblog.com/search/label/Chrome%20for%20Android

I am going to push back on this. As @thestinger states in #931, "Chrome for Android release notes don't explicitly announce vulnerabilities that were disclosed in the desktop release notes." An example of this is CVE-2021-30551, which was patched in [Chromium 91.0.4471.101](https://chromereleases.googleblog.com/2021/06/stable-channel-update-for-desktop.html). Looking at the corresponding [issue](https://bugs.chromium.org/p/chromium/issues/detail?id=1216437) on the Chromium bug tracker, it affects Android in addition to desktop variants of Chromium, but this vulnerability was not listed in the release notes of [Chromium for Android 91.0.4472.101](https://chromereleases.googleblog.com/2021/06/chrome-for-android-update_01297860997.html).

**Reply by csagan5** [reply]
@Zoraver when I posted my comment there was no new release for Android; now you can see one: https://chromereleases.googleblog.com/2021/10/chrome-for-android-update.html

Now someone should go through the [git log](https://chromium.googlesource.com/chromium/src/+log/94.0.4606.71..94.0.4606.80?pretty=fuller&n=10000) and see if the fix was also applied on the Android version, and if the corresponding issues are now de-embargoed so that one can verify whether they affected Android or not.

Until the Android release is made (after the Desktop one) there is simply no way to check this.

**Reply by Zoraver** [reply]
I misunderstood what you were trying to express; sorry about that. That said, the release that snrkl linked to is from last week and the [corresponding version of Chromium for Android](https://chromereleases.googleblog.com/2021/09/chrome-for-android-update_30.html) was released on the same day.

**Reply by csagan5** [reply]
I was looking for an Android release **after** the Desktop one; did not notice the one done on the same day.

By looking at that Android release [git log](https://chromium.googlesource.com/chromium/src/+log/94.0.4606.61..94.0.4606.71?pretty=fuller&n=10000) I see that https://crbug.com/1245578 and https://crbug.com/1251787, both still embargoed, have a fix in there. https://crbug.com/1252918 instead (about V8) does not.

At least for the former 2 issues it is safe to conclude that they do affect Android, and hopefully they will remove the embargo soon.
